.github/workflows/tests.yml

```diff
@@ -5,85 +5,241 @@ on: [push, pull_request]
+
+      # Install node and yarn in order to build the front end during packaging  
+      - name: Set Node.js 20.x
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+          cache: 'yarn'
+          cache-dependency-path: locust/webui/yarn.lock
+      - name: Install Yarn
+        run: npm install -g yarn
+
+      # Build and upload the project artifacts only once
+      - name: Build Python dist and web UI
+        run: poetry build --no-interaction
+      - name: Upload Python dist as Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: python-dist
+          path: dist/*
+
+      - name: Upload Web UI as Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: webui-dist
+          path: locust/webui/dist/*
+  
+      - name: Build UI library
+        run: yarn webui:build:lib
+      - name: Upload web UI library as Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: webui-lib-dist
+          path: locust/webui/lib
+      
+      # Set workflow metadata in one place so we can pull it out later
+      - id: set_tag
+        run: echo "tag=$(poetry version -s)" >> "$GITHUB_OUTPUT"
+      - id: set_tag_short
+        run: echo "tag_short=$(poetry version -s | cut -d '.' -f1-3)" >> "$GITHUB_OUTPUT"
+      - id: set_branch
+        run: echo "branch=${{ github.head_ref || github.ref_name }}" >> "$GITHUB_OUTPUT"
+      - id: set_is_merge_commit
+        run: echo "is_merge_commit=$( [ $(git rev-list --count $GITHUB_SHA^@) -eq 2 ] && echo 'true' || echo 'false' )" >> "$GITHUB_OUTPUT"
+      - id: set_python_version
+        run: echo "python_version=$(python -VV | sha256sum | cut -d' ' -f1)" >> "$GITHUB_OUTPUT"
+
+  #-------------------------
+  #       Testing
+  #-------------------------
+
+  tox_tests:
     name: ${{ matrix.name }}
     runs-on: ${{ matrix.os }}
+    needs: build_package
 
     strategy:
       fail-fast: false
       matrix:
         include:
-          - {name: Windows, python: '3.12', os: windows-latest, tox: fail_fast_test_main }
-          # - {name: Mac, python: '3.12', os: macos-latest, tox: fail_fast_test_main }
-          - { name: "ruff", python: "3.11", os: ubuntu-latest, tox: "ruff" }
-          - { name: "mypy", python: "3.10", os: ubuntu-latest, tox: "mypy" }
-          # run some integration tests and abort immediately if they fail, for faster feedback
-          - { name: "Linux", python: "3.12", os: ubuntu-latest, tox: fail_fast_test_main }
-          - { name: "3.12", python: "3.12", os: ubuntu-latest, tox: py312 }
-          - { name: "3.11", python: "3.11", os: ubuntu-latest, tox: py311 }
-          - { name: "3.10", python: "3.10", os: ubuntu-latest, tox: py310 }
-          - { name: "3.9", python: "3.9", os: ubuntu-latest, tox: py39 }
+          # Static analysis and utilities
+          - { name: "Verification Tests: Ruff", python: "3.11", os: ubuntu-latest, tox: "ruff" }
```

Comment:
I think you can drop the prefixes for the names. "Ruff/Mypy/Python 3.X" are specific enough, and I dont really know what a "verification test" is :P

Reply:
And if you really want to have prefixes, maybe "static analysis" is a better prefix?

Reply:
Addressed in https://github.com/locustio/locust/pull/2873/commits/7d7bfc673d7b2676d3491d8f8218510d8d1bfa49